### ISSUE TITLE
Flagging support for omnical

### ISSUE THREAD

**Issue by jpober** [HERA-Team/hera_cal#2]
It seems (surprisingly) that Omnical ignores flags in the data.  This may require editing C code to support, but it certainly seems important to do.

**Comment by jsdillon** [comment]
As it turns out, this is sort-of-partially supported in omnical using the filter_reds function: https://github.com/HERA-Team/omnical/blob/master/src/arrayinfo.py#L5-L23

When you create an info object using init_from_reds (using omnical) or aa_to_info (in heracal), you can remove baselines, antennas, or whole unique baseline groups using ex_bls, ex_ants, or ex_ubls. 

Usually it's one info object created per file to calibrate, though in theory you can create create them more often (it's slower). This means that it's fairly trivial to exclude specific baselines, not just antennas, from the calibration of a single file. Is that sufficient for now?

**Comment by bhazelton** [comment]
I think that if this is the solution, there needs to be infrastructure in hera_cal around it to take a flag array -- which is a very standard item that is handed around in all interferometric analysis packages -- and use it to generate the info objects so that users don't have to do it by hand.

**Comment by jpober** [comment]
I agree with Bryna.  Even if it's not super-graceful at the underlying code level, this has to be a basic feature, not a hack.  It would be good to benchmark how slow it is to create an info file per-time per-frequency and see if this is a viable solution.

**Comment by wenyang-li** [comment]
At least generating info per (t,f) sample in python code is unrealistic. For MWA data files, there are 384*56 time and frequency samples per file. Even if it takes only 1 second per loop, which is way too optimistic, it would still take 6 hours to calibrate 1 file.

**Comment by jpober** [comment]
Yes, put that way, it does sound prohibitively expensive.  Fundamentally, though, is there any other way to do it?  Could all times/frequencies in which there are zero flags be solved for simultaneously, and then separate equations for the rest?  Seems like that's pretty convoluted and probably wouldn't help that much.  Is the matix in the C code built up line by line?  Could the flags be applied there?

**Comment by jpober** [comment]
Definitely not convinced it's irrelevant.  Had some conversations about this with @adampbeardsley.  RTP will put in baseline-dependent flags with the watershed method.  Unless there is a unified flag mask for all baselines, omnical enables flagged samples to affect non-flagged samples by solving for antenna based gains.  Not sure how much of a problem, but definitely a possibility.

**Comment by jsdillon** [comment]
This was discussed a bit at the CHAMP bootcamp. 

As long as we're only doing RFI flags that are only a function of frequency and time (not baseline), this isn't an issue. That's the current mode of operation and the planned future mode of operation for xrfi. However, for H2CB there are plans to do flagging before baseline-dependent averaging and so there's a question of whether or not that flagging will itself be baseline-dependent. If it is, we might need this upgrade. On the other hand, we could also do this with redcal (which works better than omnical, but slower) if it is sufficiently accelerated with GPUs.
